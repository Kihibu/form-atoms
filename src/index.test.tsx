// import { renderHook } from "@testing-library/react-hooks";
import { render, screen } from "@testing-library/react";
// import userEvent from '@testing-library/user-event
import { Provider } from "jotai";
import React from "react";
import type { FieldAtom } from ".";
import { fieldAtom, useFieldAtom } from ".";

describe("useFieldAtom()", () => {
  it("should pass", () => {
    const firstNameAtom = fieldAtom({
      name: "firstName",
      value: "",
    });
    const Form = createForm(firstNameAtom);
    const result = render(
      <Provider>
        <Form />
      </Provider>
    );
    expect(result.asFragment()).toMatchSnapshot();
    expect(screen.getByRole("textbox")).toHaveAttribute("name", "firstName");
  });
});

function createForm<Value>(fieldAtom: FieldAtom<Value>) {
  return function Form() {
    const field = useFieldAtom(fieldAtom);
    console.log("wtf", field);
    // @ts-expect-error
    return <input type="text" {...field.props} />;
  };
}
