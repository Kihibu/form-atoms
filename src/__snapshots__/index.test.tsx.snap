// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFieldAtom() should pass 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    name="firstName"
    type="text"
    value=""
  />
</DocumentFragment>
`;
